### PR TITLE
Nerf the Seraphim T3 Submarine Hunter and make larger formations of it less oppressive

### LINKED ISSUE #5894: The Yathsou (T3 Submarine Hunter) has become too strong after its recent buff
The 28 % DPS buff from #5834 in combination with its very effective torpedo defense, makes creating death blobs of this unit trivial. At the moment, this unit's performance is roughly comparable to how Sniper Bots were before their various nerfs. 

|   | Yathsou | Barracuda | Vesper |
| ------------- | ------------- | ------------- | ------------- |
| HP per mass | 1.333 | 1.3 | 1.636 |
| DPS per mass | 0.121 | 0.102 | 0.082 |
| Range | 65 | 45 | 45 |
| Anti-Air | Yes | No | No |

It has more DPS/mass than both Tech 2 Sub Hunters, while also having anti-air capabilities and 20 additional range.

### PR DESCRIPTION
**Pull request by Basilisk3**
## Description of the proposed changes

**Main armament**
Damage: 290 --> 380
RateOfFire: 0.25 --> 0.22
MuzzleSalvoDelay: 0.7 --> 1
MuzzleSalvoSize: 5 --> 4
DPS: 363 --> 338

**Torpedo Defenses**
RateOfFire: 0.15 --> 0.1

## Results of the changes

- Destroyers and Torpedo Boats are more effective against the Yathsou
- Torpedo Defenses cannot be overwhelmed as quickly as before
- Torpedo Bombers are more effective
- Ground-firing remains unchanged

For further explantions see #5894 (Closes #5894)

## Checklist

- [x] Changes are documented in the changelog for the next game version (in separate PR)

### CODE REVIEW

**Comment by zqh9843** [comment]
The nerf was a bit too much imho tbh. Also considering Sera's lack of mass efficient T2 sub killer, I think nerfing the damage to 270 so it's closer to the pre-change stats(about 337.5) is way better than overdoing it.

File: units/XSS0304/XSS0304_unit.bp [hunk 1]
```diff
@@ -227,7 +227,7 @@ UnitBlueprint{
             },
             BallisticArc = "RULEUBA_None",
             CollideFriendly = false,
-            Damage = 290,
+            Damage = 240,
```
Comment (zqh9843) [inline units/XSS0304/XSS0304_unit.bp:230]: 270 maybe, revert the damage to its original stats.

**Comment by Basilisk3** [comment]
I understand your concerns, please see my comment in #5894 for further context and an explanation as to why this is necessary.

File: units/XSS0304/XSS0304_unit.bp (hunk 1 above)
Comment (Basilisk3) [reply units/XSS0304/XSS0304_unit.bp:230]: See #5894

**Review by MrRowey — APPROVED** [review]

**Pull request merged**